packages/firestore/test/unit/specs/listen_spec.test.ts

```diff
@@ -703,53 +695,103 @@ describeSpec('Listens:', [], () => {
     const query1 = Query.atPath(path('collection'));
     const query2 = Query.atPath(path('collection'));
 
-    return client(0)
-      .userListens(query1)
-      .watchAcksFull(query1, 1000)
-      .expectEvents(query1, {})
-      .client(1)
-      .userListens(query1)
-      .expectEvents(query1, {})
-      .client(0)
-      .disableNetwork()
-      .expectEvents(query1, { fromCache: true })
-      .client(1)
-      .expectEvents(query1, { fromCache: true })
-      .client(2)
-      .userListens(query1)
-      .expectEvents(query1, { fromCache: true })
-      .userListens(query2)
-      .expectEvents(query2, { fromCache: true });
+    return (
+      client(0)
+        .userListens(query1)
+        .watchAcksFull(query1, 1000)
+        .expectEvents(query1, {})
+        .client(1)
+        // Prevent client 0 from releasing its primary lease.
+        .disableNetwork()
+        .userListens(query1)
+        .expectEvents(query1, {})
+        .client(0)
+        .disableNetwork()
+        .expectEvents(query1, { fromCache: true })
+        .client(2)
+        .userListens(query1)
+        .expectEvents(query1, { fromCache: true })
+        .userListens(query2)
+        .expectEvents(query2, { fromCache: true })
+    );
   });
 
   specTest(
-    "Secondary client's online state is ignored",
+    'New client becomes primary if no client is online',
```

Comment:
"is online" => "has its network enabled"

Reply:
Done